File: src/test/java/recipestore/db/triplestore/rdfparsers/LenientTokenizerTest.java
package recipestore.db.triplestore.rdfparsers;

import org.apache.jena.riot.tokens.TokenType;
import org.apache.jena.riot.tokens.Tokenizer;
import org.hamcrest.Matchers;
import org.jooq.lambda.Seq;
import org.junit.Test;

import java.util.AbstractMap;
import java.util.List;
import java.util.stream.Collectors;

import static com.google.common.collect.Lists.newArrayList;
import static org.hamcrest.MatcherAssert.assertThat;

public class LenientTokenizerTest {


    @Test
    public void shouldTokenizeBadURI() {
        final List<TokenType> expected = newArrayList(TokenType.BNODE, TokenType.IRI, TokenType.LITERAL_LANG, TokenType.IRI
                , TokenType.DOT);

        final List<String> expectedValue = newArrayList("node2ea0be87cb5aa410715f477d8126cd8f",
                "http://schema.org/NutritionInformation/sodiumContent",
                "Sodium 480mg",
                "http://relish.com/recipes/creamy-braising-greens-soup/", null);

        final String quadWithBadURI =
                "_:node2ea0be87cb5aa410715f477d8126cd8f <http://schema.org/NutritionInformation/\\\"sodiumContent\\\"> \"Sodium 480mg\"@en-us <http://relish.com/recipes/creamy-braising-greens-soup/> .";
        final Tokenizer tokenizer = LenientTokenizer.create(quadWithBadURI);
        final List<AbstractMap.SimpleEntry<TokenType, String>> actualValues = Seq.seq(tokenizer).map(token -> new AbstractMap.SimpleEntry<>(token.getType(), token.getImage()))
                .toList();


        assertThat("Contains TokenTypes in the correct order", actualValues.stream().map(val -> val.getKey()).collect(Collectors.toList()), Matchers.equalTo(expected));
        assertThat("Contains values in the correct order", actualValues.stream().map(val -> val.getValue()).collect(Collectors.toList()), Matchers.equalTo(expectedValue));


    }

}

File: src/test/scala/recipestore/db/triplestore/rdfparsers/LenientTokenizerTest.scala
package recipestore.db.triplestore.rdfparsers

import org.apache.jena.riot.tokens.{TokenType, Tokenizer}
import org.scalatest.{FunSuite, Matchers}

import scala.collection.JavaConverters._

class LenientTokenizerTest extends FunSuite with Matchers {
  test("Should tokenize bad uri") {
    val expected: List[TokenType] = List(TokenType.BNODE, TokenType.IRI, TokenType.LITERAL_LANG, TokenType.IRI, TokenType.DOT)
    val expectedValue: List[String] = List("node2ea0be87cb5aa410715f477d8126cd8f", "http://schema.org/NutritionInformation/sodiumContent", "Sodium 480mg", "http://relish.com/recipes/creamy-braising-greens-soup/", null)
    val quadWithBadURI: String = "_:node2ea0be87cb5aa410715f477d8126cd8f <http://schema.org/NutritionInformation/\\\"sodiumContent\\\"> \"Sodium 480mg\"@en-us <http://relish.com/recipes/creamy-braising-greens-soup/> ."
    val tokenizer: Tokenizer = LenientTokenizer.create(quadWithBadURI)
    val actualValues: Seq[(TokenType, String)] = tokenizer.asScala.toSeq.map(token => (token.getType(), token.getImage()))
    actualValues.map(v => v._1) should be equals expected
    actualValues.map(v => v._2) should be equals expectedValue
  }
}